src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -1587,6 +1601,32 @@ public int getRemainingOperations() {
         }
     }
 
+    public void tryExtractItems() {
+        if (!isMissingMode || this.waitingForMissing.isEmpty()) return;
+        if (countToTryExtractItems > 1200) {
+            countToTryExtractItems = 0;
+            for (IAEItemStack waitingForItem : this.waitingForMissing) {
+                final IGrid grid = this.getGrid();
+                if (grid != null) {
+                    final IStorageGrid pg = grid.getCache(IStorageGrid.class);
+                    if (pg != null) {
+                        IAEItemStack extractedItems = pg.getItemInventory()
+                                .extractItems(waitingForItem, Actionable.MODULATE, this.machineSrc);
+                        if (extractedItems != null) {
+                            IAEStack notInjected = injectItems(extractedItems, Actionable.MODULATE, this.machineSrc);
+                            if (notInjected != null) { // not sure if this even need, but still
+                                pg.getItemInventory()
+                                        .injectItems((IAEItemStack) notInjected, Actionable.MODULATE, this.machineSrc);
+                            }
+                        }
```

Comment:
If there is concern about not being able to inject the full amount, i would first do simulate extract and inject to get the proper amount you can use, then do them again on modulate. This is more performance intensive, but if its needed its needed. May be worth throwing a logger call in this inner-most if-statement to see if its ever encountered in a real-world scenario

Reply:
i just tested, cpu doesn't have any byte limitations for storage, so storage overflow shouldn't happen at all. And it request exactly how much cpu need. I added log which will notify players in chat, i guess after testing "overflow" part can be fully removed.